### PR TITLE
Upgrade Vault ruby gem to 0.10.1

### PR DESCRIPTION
**Pull request by jonmoter**
According to the [vault-ruby releases](https://github.com/hashicorp/vault-ruby/releases), we're over a year old.  So let's upgrade to a newer version of the gem.

(I was unable to run tests locally, so I'm hoping Travis will have more luck.)

@zendesk/paas, @zendesk/infraops

### CODE REVIEW

File: .ruby-version [hunk 1]
```diff
@@ -1 +1 @@
-2.2.4
+2.4.1
```
Comment (grosser) [inline .ruby-version]: bump dockerfile too ?

Comment (jonmoter) [reply .ruby-version]: Bumped!

**Comment by jonmoter** [comment]
Hmm, test failing...

```
 1) Error:
CRON#test_0001_can run the cron job:
Errno::EADDRINUSE: Address in use - bind(2) for 127.0.0.1:8200
```

Might be due to an updated version of webmock

**Comment by jonmoter** [comment]
Closing in favor of #50, which doesn't muck with ruby versions or webmock.

**Pull request closed without merging**